client/src/components/MultiJurisdictionAudit/AASetup/Contests/HybridContestForm.test.tsx

```diff
@@ -0,0 +1,270 @@
+    },
+  ],
+}
+
+const { nextStage, prevStage } = relativeStages('target-contests')
+
+const render = (isTargeted: boolean = true) =>
+  renderWithRouter(
+    <Route path="/election/:electionId/setup">
+      <Contests
+        auditType="HYBRID"
+        locked={false}
+        isTargeted={isTargeted}
+        nextStage={nextStage}
+        prevStage={prevStage}
+      />
+    </Route>,
+    { route: '/election/1/setup' }
+  )
+
+const apiCalls = {
+  getStandardizedContests: {
+    url: '/api/election/1/standardized-contests',
+    response: [
+      {
+        name: 'Contest 1.\'"', // Make sure dots and quotes in the name work
+        jurisdictionIds: ['jurisdiction-id-1', 'jurisdiction-id-2'],
+      },
+      {
+        name: 'Contest 2',
+        jurisdictionIds: ['jurisdiction-id-1'],
+      },
+      { name: 'Contest 3', jurisdictionIds: ['jurisdiction-id-2'] },
+    ],
+  },
+  getContests: (contests: IContest[]) => ({
+    url: '/api/election/1/contest',
+    response: { contests },
+  }),
+  putContests: (contests: INewContest[]) => ({
+    url: '/api/election/1/contest',
+    options: {
+      method: 'PUT',
+      body: JSON.stringify(contests),
+      headers: { 'Content-Type': 'application/json' },
+    },
+    response: { status: 'ok' },
+  }),
+  submitContests: (mockContest: object) => ({
+    url: '/api/election/1/contest',
+    options: {
+      method: 'PUT',
+      body: JSON.stringify(mockContest),
+      headers: { 'Content-Type': 'application/json' },
+    },
+    response: { status: 'ok' },
+  }),
+}
+
+jest.mock('uuidv4', () => {
+  let id = 0
+  return () => {
+    id += 1
+    return id.toString()
+  }
+})
+
+function typeInto(input: Element, value: string): void {
+  // TODO: do more, like `focusIn`, `focusOut`, `input`, etc?
+  fireEvent.focus(input)
+  fireEvent.change(input, { target: { value } })
+  fireEvent.blur(input)
+}
+
+describe('Audit Setup > Contests (Hybrid)', () => {
+  const newContest = [
+    {
+      id: '1',
+      name: 'Contest 1.\'"',
+      isTargeted: true,
+      totalBallotsCast: 30,
+      numWinners: 1,
+      votesAllowed: 1,
+      jurisdictionIds: ['jurisdiction-id-1', 'jurisdiction-id-2'], // adding these makes the test fail, hence the functionality needs to be changed?
+      choices: [
+        {
+          id: '2',
+          name: 'Choice One',
+          numVotes: 10,
+        },
+        {
+          id: '3',
+          name: 'Choice Two',
+          numVotes: 20,
+        },
+      ],
+    },
+  ]
+
+  it('Audit Setup > Contests', async () => {
+    const expectedCalls = [
+      apiCalls.getContests([]),
+      aaApiCalls.getJurisdictions,
+      apiCalls.getStandardizedContests,
+    ]
+    await withMockFetch(expectedCalls, async () => {
+      const { container } = render()
+      await screen.findByRole('heading', { name: 'Target Contests' })
+      expect(container).toMatchSnapshot()
+    })
+  })
+
+  it('is able to submit form successfully', async () => {
```

Comment:
i'm curious if you've tested that the form works after you've submitted contests and you come back to the edit them. i could see that being a bit hairy with this new dropdown

Reply:
Good point, I'll double check that.